src/lib.rs

```diff
@@ -379,6 +382,27 @@ impl OpenatDirExt for openat::Dir {
         .map_err(map_nix_error))
     }
 
+    fn set_mode<P: openat::AsPath>(&self, p: P, mode: libc::mode_t) -> io::Result<()> {
+        use nix::sys::stat::{fchmodat, FchmodatFlags, Mode};
+
+        let path = p
+            .to_path()
+            .ok_or_else(|| io::Error::new(io::ErrorKind::Other, "null byte in path"))?;
+        let perms = Mode::from_bits_truncate(mode);
+
+        // NOTE(lucab): `AT_SYMLINK_NOFOLLOW` exists but it is not really implemented
+        // and short-circuits to `ENOTSUP`.
```

Comment:
After going through this I've additionally discovered that this behavior changed in a very recent glibc release (see https://github.com/coreos/rpm-ostree/pull/2789 for references).

I'm thinking that it may be better to detect the `is_link` case explicitly and short-circuit that.

Reply:
I've pushed a second commit on top which goes in that direction. Let me know if it is reasonable.